### ISSUE TITLE
Client API : Asynchronous calls timeout management

### ISSUE THREAD

**Issue by tango-controls-bot** [tango-controls/cppTango#149]
We are using the DeviceAttribute *DeviceProxy::read_attribute_reply(long id,long call_timeout) method in a client application (in fact a Tango server) and experienced the following scenario. 

If I set the timeout parameter, in the method (ex read_attribute_reply(id,100) , I have a fixed overhead of 20 ms to get my answer, and in the case of my application , this overhead is not applicable . 
In fact, if the timeout parameter is set,  if we look into the current implementation (file proxy_asyn.cpp), the implementation algorithm is the following.:

..
#ifdef _TG_WINDOWS_
Sleep(20);
#else
nanosleep(&to_wait,&inter);
#endif
if (req.request-&gt;poll_response() == true)
{
break;
}

In other words, the current implementation always wait 20 ms, and then tries to get the answer, rather than trying to get the answer and then sleep before the next retry. 

The implementation should be :
- try to get the answer
- then start the retrying loop 

A quick look into this file , shows that other methods (DeviceData Connection::command_inout_reply(long id,long call_timeout))  have the same "poor performance" implementation 

Manu could you help ? 

Reported by: buteau

Original Ticket: [tango-cs/bugs/529](https://sourceforge.net/p/tango-cs/bugs/529)
